### DISCUSSION TITLE
Factor variables cause fatal error and c-index

### DISCUSSION THREAD

**Discussion by RalbertJastein** [kogalur/randomForestSRC#430] [Q&A]
Hi,

I am new to the Randomforestsrc package and I'm runnning into some problems.

Firstly, Im trying to run a classification model with both binary factor and numeric features. However, when I change the binary factor variables (coded as 0 and 1) to factors, R encounters a "fatal error" and shuts down when I run the rfsrc function.. When i run the model with the factor variables coded as numeric everything works fine. What could be the problem? My dataset contains approx. 2300 observations and 40 predictors. 

Also, Im getting a c-index below 0.5, is this c-index = 1-AUC?

Thanks for your help!

**Comment by ishwaran** [comment]
Regarding your second question, C-index = 1 - Harrell's concordance.  Therefore smaller values indicates a smaller error rate and better performance.

Regarding your first question.  There is no reason for this to happen with mixed data in our experience.  In order to proceed further please provide a reproducible example.

**Reply by jeremywzh** [reply]
I encountered the same problem. For my data set, if I use factor,  R encounters a "fatal error" and shuts down when I run the rfsrc function. But if I convert factor to integer, it works fine.

It seems that this issue depends on dataset. For the data set: data(peakVO2, package = "randomForestSRC"), it works fine if I convert "male" variable into a factor.

Do you have any idea what the problem is? There is no missing value in my dataset.

**Reply by jeremywzh** [reply]
I found out why. It works fine if the data is a data frame. Once I use as_tibble() to convert the data.frame to a tibble, it crashes on factors.

**Comment by ishwaran** [comment]
Yes, you have to use a data frame for `rfsrc` and you cannot use a tibble.
